reaper/reaper.go

```diff
@@ -632,162 +631,96 @@ func allReapables() (map[string][]reaperevents.Reapable, []reaperevents.Reapable
+// matchesFilters applies the relevant filter groups to a filterable
+func matchesFilters(filterable filters.Filterable) bool {
 	// recover from potential panics caused by malformed filters
 	defer func() {
 		if r := recover(); r != nil {
-			log.Error("Recovered in applyFilters with panic: ", r)
+			log.Error("Recovered in matchesFilters with panic: ", r)
 		}
 	}()
 
-	whitelistedCount := make(map[string]int)
-
-	var gs []reaperevents.Reapable
-	for _, filterable := range filterables {
-		var groups map[string]filters.FilterGroup
-		var stat string
-		switch filterable.(type) {
-		case *reaperaws.Instance:
-			stat = "reaper.instances.whitelistedCount"
-			groups = config.Instances.FilterGroups
-		case *reaperaws.AutoScalingGroup:
-			stat = "reaper.asgs.whitelistedCount"
-			groups = config.AutoScalingGroups.FilterGroups
-		case *reaperaws.Cloudformation:
-			stat = "reaper.cloudformations.whitelistedCount"
-			groups = config.Cloudformations.FilterGroups
-		case *reaperaws.SecurityGroup:
-			stat = "reaper.securitygroups.whitelistedCount"
-			groups = config.SecurityGroups.FilterGroups
-		case *reaperaws.Volume:
-			stat = "reaper.volumes.whitelistedCount"
-			groups = config.Volumes.FilterGroups
-		default:
-			log.Warning("You probably screwed up and need to make sure applyFilters works!")
-			return []reaperevents.Reapable{}
-		}
-
-		matched := false
-
-		// if there are no filters groups defined, default to a match
-		if len(groups) == 0 {
-			matched = true
-		}
-
-		// if there are no filters, default to a match
-		noFilters := true
-		for _, group := range groups {
-			if len(group) != 0 {
-				// set to false if any are non-zero length
-				noFilters = false
-			}
-		}
-		if noFilters {
-			matched = true
-		}
-
-		for name, group := range groups {
-			didMatch := filters.ApplyFilters(filterable, group)
-			if didMatch {
-				matched = true
-				filterable.AddFilterGroup(name, group)
-			}
-		}
+	var groups map[string]filters.FilterGroup
+	switch filterable.(type) {
+	case *reaperaws.Instance:
+		groups = config.Instances.FilterGroups
+	case *reaperaws.AutoScalingGroup:
+		groups = config.AutoScalingGroups.FilterGroups
+	case *reaperaws.Cloudformation:
+		groups = config.Cloudformations.FilterGroups
+	case *reaperaws.SecurityGroup:
+		groups = config.SecurityGroups.FilterGroups
+	case *reaperaws.Volume:
+		groups = config.Volumes.FilterGroups
+	default:
+		log.Warning("You probably screwed up and need to make sure matchesFilters works!")
+		return false
+	}
 
-		// whitelist filter
-		if filterable.Filter(*filters.NewFilter("Tagged", []string{config.WhitelistTag})) {
-			// if the filterable matches this filter, then
-			// it should be whitelisted, aka not matched
-			matched = false
-			whitelistedCount[stat]++
-		}
+	matched := false
 
-		if matched {
-			gs = append(gs, filterable)
-		}
+	// if there are no filters groups defined, default to a match
+	if len(groups) == 0 {
+		matched = true
```

Comment:
Can this function be refactored to return early instead of maintaining these boolean variables? For example, instead of
```if len(groups) ==0 {
   matched = true
}```

can you just do
``` if len(groups) == 0 {
return true
}```?